kroxylicious-kms-provider-hashicorp-vault/src/test/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultKmsTest.java

```diff
@@ -50,6 +55,44 @@ void testRequestTimeoutWaitingForHeaders() {
         httpServer.stop(0);
     }
 
+    static Stream<Arguments> legalEnginePaths() {
+        var uri = URI.create("https://localhost:1234");
+        return Stream.of(
+                Arguments.of("no path", uri, "/v1/transit/"),
+                Arguments.of("slash only", uri.resolve("/"), "/v1/transit/"),
+                Arguments.of("basic", uri.resolve("v1/transit"), "/v1/transit/"),
+                Arguments.of("trailing slash", uri.resolve("v1/transit/"), "/v1/transit/"),
+                Arguments.of("single namespace", uri.resolve("v1/ns1/transit"), "/v1/ns1/transit/"),
+                Arguments.of("many namespaces", uri.resolve("v1/ns1/ns2/transit"), "/v1/ns1/ns2/transit/"),
+                Arguments.of("non standard engine name", uri.resolve("v1/mytransit"), "/v1/mytransit/"));
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource
+    void legalEnginePaths(String name, URI uri, String expected) {
+        var kms = new VaultKms(uri, "token", Duration.ofMillis(500), null);
+        assertThat(kms.getEngineUri())
+                .extracting(URI::getPath)
+                .isEqualTo(expected);
+
+    }
+
+    static Stream<Arguments> illegalEnginePaths() {
+        var uri = URI.create("https://localhost:1234");
+        return Stream.of(
+                Arguments.of("unrecognized version", uri.resolve("v999/transit")),
+                Arguments.of("missing engine", uri.resolve("v1")),
+                Arguments.of("empty engine", uri.resolve("v1/")));
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource
+    void illegalEnginePaths(String name, URI uri) {
```

Comment:
Similar to the legalEnginePaths above, I'm not a fan of the name.

N.B the name makes sense for the argument source!